src/include/duckdb/main/profiling_node.hpp

```diff
@@ -0,0 +1,86 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/main/profiling_node.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/common.hpp"
+#include "duckdb/common/deque.hpp"
+#include "duckdb/common/enums/profiler_format.hpp"
+#include "duckdb/common/pair.hpp"
+#include "duckdb/common/profiler.hpp"
+#include "duckdb/common/reference_map.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/common/unordered_map.hpp"
+#include "duckdb/common/winapi.hpp"
+#include "duckdb/execution/expression_executor_state.hpp"
+#include "duckdb/execution/physical_operator.hpp"
+#include "duckdb/main/profiling_info.hpp"
+
+namespace duckdb {
+
+enum class ProfilingNodeType : uint8_t { QUERY_ROOT, OPERATOR };
+
+class QueryProfilingNode;
+
+// Recursive tree that mirrors the operator tree
+class ProfilingNode {
```

Comment:
Looking at this file with `profiling_info-c[api].cpp`, I'd prefer if we moved some of the logic into this file and made some fields private (if possible). That way, we are also exposing the functionality here.

Private fields:
- `profiling_info`
- `children`

Functions to add here (and then call them from the C API):
- `ProfilingInfo &GetProfilingInfo();`
- `ProfilingNode &GetChild(idx_t index);`